tensorflow/core/kernels/dml_batch_to_space_op.cc

```diff
@@ -191,6 +205,29 @@ class BaseBatchToSpaceInitHelper : public InitializationHelper {
     return false;
   }
 
+  absl::optional<int> GetForwardableInputIndex(
+      OpKernelContext* ctx, absl::Span<const TensorShape> output_shapes,
+      int outputIndex) const override {
+    // For batchtospace, we can only forward input 0 to output 0
+    if (outputIndex != 0) {
+      return {};
+    }
+
+    const Tensor& input =
+        ctx->input_is_ref(0) ? ctx->mutable_input(0, false) : ctx->input(0);
```

Comment:
You don't need the `ctx->input_is_ref(0)` check here since we know that the batch/space ops don't have ref inputs.